HelpMyStreetFE/HelpMyStreetFE/src/js/requesthelp/requesthelp.js

```diff
@@ -0,0 +1,155 @@
+
+		try {
+			const resp = await fetch(`/api/postcode/checkCoverage/${postcode}`);
+			console.log(resp);
+			if (resp.ok) {						
+				const response = await resp.json();
+			
+				console.log(response);	
+					// also in response now is volunteerCount and championCount
+				if (response.postCodeResponse.hasContent && response.postCodeResponse.isSuccessful) {
+					$("select[name=address_selector]").html(
+						response.postCodeResponse.content.addressDetails.reduce((acc, cur, i) => {
+							const text = Object.keys(cur).reduce((tAcc, tCur) => {
+								if (cur[tCur] != null) {
+									tAcc += tAcc === "" ? "" : ", ";
+									tAcc += `${cur[tCur]}`;
+								}
+
+								return tAcc;
+							}, "");
+
+							acc += `<option value="${i}">${text}</option>`;
+							return acc;
+						}, '<option value="" selected disabled hidden>Choose here</option>')
+					);
+
+					$("#address_selector").slideDown();
+					$("select[name=address_selector]").on("change", function () {
+						const id = $(this).children("option:selected").val();
+
+						const address = response.postCodeResponse.content.addressDetails[id];
+
+						$("input[name=address_line_1]").val(address.addressLine1);
+						$("input[name=address_line_2]").val(address.addressLine2);
+						$("input[name=city]").val(address.locality);
+						$("input[name=postcode]").val(address.postcode);
+						$(".expander").slideDown();
+						$(".expanderDetails").slideDown();
+						
+
+					});
+				}
+			}
+		} catch (ex) {
+			console.error(ex);
+		}
+		buttonUnload($(this));
+	});
+
+	$("#requesthelp_form").on("submit", function (event) {
+		$(".expander").slideDown();
+
+		event.preventDefault(); 
+
+		let validHelpNeeded = true;
+
+		var obj = $(this).serializeArray().reduce(function (acc, cur) {
+			acc[cur.name] = cur.value;
+			return acc;
+		}, {});
+		$(".error").hide();
+		
+		if (!obj["helpNeededArray"]) {
+			validHelpNeeded = false;
+			$("#helpNeeded-error").show();
+		}
+
+		const valid = validateFormData($(this), {
+			firstname: (v) => v !== "" || "Please enter a first name",
+			lastname: (v) => v !== "" || "Please enter a last name",
+			email: (v) => v !== "" || "Please enter an email address",
+			postcode: (v) => v !== "" || "Please enter a postcode",
+			phonenumber: (v) =>
+				v == "" ||
+				((v.replace(" ", "").length === 10 || v.replace(" ", "").length === 11) && v[0] === "0") ||
+				"Please enter a valid phone number",
+			city: (v) =>
+				(v.length > 2) ||
+				"Please enter a valid city",
+			address_line_1: (v) =>
+				(v.length > 2) ||
+				"Please enter a valid first line of your address",
+		});
+
+		let validForm = (valid && validHelpNeeded);
+
+		let emailValid = true;
+		let emailInput = $("input[name='email']");
+		let emailInputError = emailInput.find("~ .error");
+
+		const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
Email Valid Regex test  can should be used inside the validateformData function. 
Take a look at stepone.js for the password field, thats an example of using a regex match.
![image](https://user-images.githubusercontent.com/25431693/79737546-63211e00-82f3-11ea-81ce-9d41ce090c17.png)